src/python/pants/engine/target.py

```diff
@@ -487,33 +489,27 @@ def prefix_glob_with_address(self, glob: str) -> str:
 
 
 @dataclass(frozen=True)
-class SourcesRequest:
+class HydrateSourcesRequest:
     field: Sources
 
 
 @dataclass(frozen=True)
-class SourcesResult:
+class HydratedSources:
     snapshot: Snapshot
 
-    # TODO: do we want to support the `extension` parameter from Target.has_sources()? In V1,
-    # it's used to distinguish between Java vs. Scala files. For now, we should leave it off to
-    # keep things as simple as possible, but we may want to add it in the future.
-    def has_sources(self) -> bool:
-        return bool(self.snapshot.files)
```

Comment:
This wasn't being used. For now, it's simple to say in downstream rules `if not hydrated_sources.snapshot.files`, which is what we end up doing in this PR and the followup.

We can always add this back later. It's better to keep our APIs as small as possible for now.